util/filter_bench.cc

```diff
@@ -75,21 +84,31 @@ using rocksdb::Slice;
 using rocksdb::mock::MockBlockBasedTableTester;
 
 struct KeyMaker {
-  KeyMaker(size_t size)
-      : data_(new char[size]),
-        slice_(data_.get(), size),
-        vals_(reinterpret_cast<uint32_t *>(data_.get())) {
-    assert(size >= 8);
-    memset(data_.get(), 0, size);
+  KeyMaker(size_t avg_size)
+      : smallest_size_(avg_size -
+                       (FLAGS_vary_key_size_log2_interval >= 30 ? 2 : 0)),
+        n_vals_((avg_size + 11) / 4),  // pad to vary key size and alignment
+        vals_(new uint32_t[n_vals_]),
+        data_(reinterpret_cast<char *>(vals_.get())) {
+    memset(data_, 0, n_vals_ * 4);
+    assert(smallest_size_ > 8);
   }
-  std::unique_ptr<char[]> data_;
-  Slice slice_;
-  uint32_t *vals_;
+  size_t smallest_size_;
+  size_t n_vals_;
+  std::unique_ptr<uint32_t[]> vals_;
+  char *data_;
 
   Slice Get(uint32_t filter_num, uint32_t val_num) {
-    vals_[0] = filter_num + val_num;
-    vals_[1] = val_num;
-    return slice_;
+    size_t start = FLAGS_vary_key_alignment ? val_num % 4 : 0;
+    size_t len = smallest_size_;
+    if (FLAGS_vary_key_size_log2_interval < 30) {
+      len += fastrange32(
+          (val_num >> FLAGS_vary_key_size_log2_interval) * 1234567891, 5);
+    }
+    vals_[0] = val_num;
+    vals_[1] = filter_num + val_num;
+    vals_[2] = val_num;
```

Comment:
Somehow I lost what those `vals_` are for and I didn't quickly figure out by reading the code. Can you add some comments on what vals_ are used for to make it easier for review?